File: src/main/scala/unwrittenfun/minecraft/unwrittenblocks/client/gui/GuiSidesButton.scala
package unwrittenfun.minecraft.unwrittenblocks.client.gui

import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiButton

/**
 * Project: UnwrittenBlocks
 * Author: UnwrittenFun 
 * Created: 14/08/2014.
 */
class GuiSidesButton(id: Int, _side: Int, _io: Int, x: Int, y: Int) extends GuiButton(id, x, y, 16, 16, "") {
  var io: Int = _io
  var side: Int = _side

  override def drawButton(minecraft: Minecraft, mX: Int, mY: Int) {
//    super.drawButton(minecraft, mX, mY)

    Minecraft.getMinecraft.renderEngine.bindTexture(GuiTextures.darkInfuser)

    if ((io & 1) == 1) drawTexturedModalRect(x, y, 176, 25, 15, 15)
    if ((io & 2) == 2) drawTexturedModalRect(x + 1, y + 1, 176, 40, 15, 15)
  }

  def buttonClicked() {
    io += 1
    if (io > 3) io = 0
  }
}

File: src/main/java/unwrittenfun/minecraft/unwrittenblocks/client/gui/components/GuiSidesButton.java
package unwrittenfun.minecraft.unwrittenblocks.client.gui.components;

import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiButton;
import unwrittenfun.minecraft.unwrittenblocks.client.gui.GuiDarkInfuser;
import unwrittenfun.minecraft.unwrittenblocks.common.tileEntities.TEConfigurableIO;

/**
 * Project: UnwrittenBlocks
 * Author: UnwrittenFun
 * Created: 06/11/2014.
 */
public class GuiSidesButton extends GuiButton {
  public int io, side;

  public GuiSidesButton(int id, int side, int io, int x, int y) {
    super(id, x, y, 16, 16, "");

    this.io = io;
    this.side = side;
  }

  @Override
  public void drawButton(Minecraft minecraft, int mX, int mY) {
    minecraft.renderEngine.bindTexture(GuiDarkInfuser.texture); //TODO: Generify texture

    if ((io & 1) == 1) drawTexturedModalRect(xPosition, yPosition, 176, 25, 15, 15);
    if ((io & 2) == 2) drawTexturedModalRect(xPosition + 1, yPosition + 1, 176, 40, 15, 15);
  }

  public void buttonClicked(TEConfigurableIO configurableIO) {
    io += 1;
    if (io > 3) io = 0;

    configurableIO.setSideIO(side, io);
  }
}
